### ISSUE TITLE
Visual studio prompt for a Password. Not in Te3 so no data.

### ISSUE THREAD

**Issue by jeroenvisser67** [TabularEditor/TabularEditor3#944]
I migrate from visual studio to Tabulare Editor. Bim File in seperate directory. In visual studio when i Open a table for data or i deploy i have to give de password for a user (data source). Here i don't have this. With data preview i don't get data. Only the fields. And on the server i can't proces de tables because there is no password. The fault i get is that the user can 't reach the sql server database. Wat can i do?? For the workspace i used integrated security and the user i mentioned . Nothing works?? Wat can i do?

No Password no data. But i don't get a fault message or a prompt for a Password like Visual studio.

**Comment by otykier** [comment]
Hi @jeroenvisser67 
What type of data source does your model use? Is it a ProviderDataSource (with a Connection String property) or a StructuredDataSource (with separate properties for server, database, etc.)?

It should still be possible to add credentials to the data source after deployment - for example using SSMS.

**Comment by jeroenvisser67** [comment]
Hi Daniel,

This is the beginning of the bim.file i open.  I ajusted the names ofcourse. This is the Datasource that gives trouble.

{
  "name": "SemanticModel",
  "compatibilityLevel": 1200,
  "model": {
    "culture": "en-US",
    "dataSources": [
      {
        "name": "aebpmis03 AMIS",
        "connectionString": "Provider=SQLNCLI11;Data Source=orsodat01\\instance1;Persist Security Info=true;User ID=userread;Packet Size=4096;Initial Catalog=AMOX",
        "impersonationMode": "impersonateServiceAccount",
        "annotations": [
          {
            "name": "ConnectionEditUISource",
            "value": "SqlServer"
          }
        ]
      },

After i openen the tmuo file looks like this. 

{
  "UseWorkspace": true,
  "WorkspaceConnection": "provider=MSOLAP;data source=orsossas01\\instance1",
  "WorkspaceDatabase": "TabularEditor3"
}

From there i try to process the data on the server. After typing in the password. Works 1 time then errors. From Analysis server on the server the can't be reached or after i filled in the password. Can't initialize the provider.
Refresh data or preview data from within Te3  gives no faults and no data . Only the fieldnames.

**Comment by jeroenvisser67** [comment]
@otykier can you answer my question?

**Comment by otykier** [comment]
@jeroenvisser67 apologies for the late response.

While the model is open in TE3, try to add your SQL password to the data source connection string:

![image](https://github.com/TabularEditor/TabularEditor3/assets/8976200/89f3864a-81c1-4eb3-929f-7b9ac67391fa)

Then hit Save (Ctrl+S). Now, try to perform a refresh (through Tabular Editor).

Also, please confirm that after hitting Save, your .tmuo file now also contains a "DataSourceOverrides" section:

<img width="463" alt="image" src="https://github.com/TabularEditor/TabularEditor3/assets/8976200/63885874-5dea-4275-b772-1a59d1990161">

**Comment by jeroenvisser67** [comment]
@otykier I work with a company where they use Visual Studio ssas tabular model. They only want to use 1 model for the whole company. Visual Studio , ssas (1 model) and Github . the source control is a nightmare. So we do a pilot with tabular editor 3 and the save to folder (splitting files option) to solve this problem. We keep the source control still in Visual Studio. It is now working. I did wat you suggested and I saved the password also on the server (the same way) and i don't deploy the data source. Now it is working. Thank you!
